src/renderer/atlas/BackendD3D.cpp

```diff
@@ -1627,9 +1628,19 @@ void BackendD3D::_drawCursorBackground(const RenderingPayload& p)
             static_cast<u16>(p.s->font->cellSize.x * (x1 - x0)),
             p.s->font->cellSize.y,
         };
-        const auto isInverted = cursorColor == 0xffffffff;
-        const auto background = isInverted ? bg ^ 0xc0c0c0 : cursorColor;
-        const auto foreground = isInverted ? 0 : bg;
+        auto background = cursorColor;
+        auto foreground = bg;
+
+        if (cursorColor == 0xffffffff)
+        {
+            background = bg ^ 0xffffff;
+            foreground = 0xffffffff;
+        }
+
+        // The legacy console used to invert colors by just doing `bg ^ 0xc0c0c0`. This resulted
+        // in a minimum squared distance of just 0.029195 across all possible color combinations.

```

Comment:
so this ensures that *all cursor foregrounds* on all colors are visible?

Reply:
It should! ...in theory at least.